CorsixTH/Lua/entities/machine.lua

```diff
@@ -98,7 +98,8 @@ local function setSmoke(self, isSmoking)
       -- Note: Set the location of the smoke to that of the machine
       -- rather than setting the machine to the parent so that the smoke
       -- doesn't get hidden with the machine during use
-      self.smokeInfo:setTile(getSmokeTile(self))
+      local map, x, y = getSmokeTile(self)
+      self.smokeInfo:setTile(map, x, y, self:getSmokeDrawingLayer())
```

Comment:
Out of interest, you change the layer 10 below. Why not just use 10 to start with?
Though I would point out layer 10 means you draw on top of side objects (good); but also floating dollars (not so good).

Though I'm not sure what the `2` does in the lower line.

Reply:
This is no less different from how it was before anyway.